src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs

```diff
@@ -82,17 +229,231 @@ private interface IMetaDataDispenser
             // Don't need any other methods.
         }
 
+#if NET6_0_OR_GREATER
+        private sealed class MetaDataDispenserWrapperCache : ComWrappers
+        {
+            public static readonly MetaDataDispenserWrapperCache Instance = new MetaDataDispenserWrapperCache();
+            private MetaDataDispenserWrapperCache() { }
+
+            protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count) => throw new NotImplementedException();
+            protected override object CreateObject(IntPtr externalComObject, CreateObjectFlags flags)
+            {
+                Debug.Assert(flags == CreateObjectFlags.UniqueInstance);
+
+                return new MetaDataDispenserRcw(externalComObject);
+            }
+            protected override void ReleaseObjects(IEnumerable objects) => throw new NotImplementedException();
+
+            public sealed unsafe record MetaDataDispenserRcw(IntPtr Inst) : IMetaDataDispenser
+            {
+                private bool _disposed = false;
+
+                public void DefineScope_Placeholder() => throw new NotImplementedException();
+
+                /// <remarks>
+                /// <paramref="punk" /> is simply a boxed IntPtr, because we don't need an RCW.
+                /// </remarks>
+                public int OpenScope(string szScope, int dwOpenFlags, ref Guid riid, out object punk)
+                {
+                    var func = (delegate* unmanaged<IntPtr, char*, int, Guid*, IntPtr*, int>)(*(*(void***)Inst + 4));
```

Comment:
I'll stop commenting, but I think each vtable entry we are using should be explicitly documented as above.

Reply:
Feels kind of redundant. Any way I could deliver the same info, but inline?

Reply:
Maybe creating an enum and naming the constant would be more readable for interfaces with a lot of methods.